src/main/kotlin/org/axonframework/intellij/ide/plugin/usage/AggregateAnnotator.kt

```diff
@@ -0,0 +1,44 @@
+package org.axonframework.intellij.ide.plugin.usage
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.psi.PsiElement
+import org.axonframework.intellij.ide.plugin.api.AxonAnnotation
+import org.axonframework.intellij.ide.plugin.util.isAggregate
+import org.jetbrains.uast.UClass
+import org.jetbrains.uast.UIdentifier
+import org.jetbrains.uast.toUElement
+
+/**
+ * Shows a warning on classes annotated with `@Aggregate` when:
+ * - They have no zero-argument constructor. This constructor is required by Axon Framework to construct the aggregate.
+ * - They have no AggregateIdentifier field. This is required by Axon Framework.
+ *
+ */
+class AggregateAnnotator : Annotator {
```

Comment:
Curious: `Annotator`s from Intellij are intended to annotate the code with warnings?

Reply:
Yes, it will show a red squiggly line (or yellow if warning) with a text